core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1253,6 +1257,10 @@ private void wakeUpSweeper(String workflowId) {
 
 			if(!outcome.tasksToBeUpdated.isEmpty() || !outcome.tasksToBeScheduled.isEmpty()) {
 				edao.updateTasks(tasksToBeUpdated);
+				boolean shallResetTags = tasksToBeUpdated.stream().anyMatch(t -> t.isTerminal() && t.shallResetTags());
```

Comment:
This is the main logic snippet of this ticket. If the task requires to do reset workflow tags (which involed into redelivery decision), it shall be reset upon task terminal status (completed/failed).

Reply:
Once workflow reached the task (wait supported for now) which requires clearing the tags at workflow level, the system resets the tags at workflow level and the workflow no longer matches "Redelivery neeeded" criteria.